### ISSUE TITLE
Account system is now availabe in our project

### ISSUE THREAD

**Issue by bill668** [tvledesign/ProjectManager#2]
1. now current user can only view or edit his own project, there is no way(I hope) user can see all List elements or add something on it.
2. I move some function to server/text.js  but router seems didn't work properly at server side.
3. There is a small bug of Done method. When user edit his project, it automatically set it as "incomplete"
4. We still need to think about Admin account method

**Comment by cmojarro** [comment]
I am testing some elemets right now

**Comment by tvledesign** [comment]
I'll check out the code tomorrow. (well in the morning / afternoon I should say)
But how is it going so far? Any troubles beside what was mentioned above?

**Comment by bill668** [comment]
Cesar said that the sharelist and homework helper can't merge very well.
Right now there a two bugs(router and done method), and the working method of it is like:
1. client: hey I have some information send to server, and right now the current user is "Ming"

2. server: OK I got the information, now let's  create a new list, in the new list there is a element call : created by_____,  and I will put Ming at here. Then I will search my database and find all project which has element: created by AAA. Then I will sent those project back to client

3. client: Great I have some list from server, now let me put it on the screen so user can see it. 

If we want to create a admin method, we better create a if statment in the server. like:

Meteor.publish('homeWorkList', function(){
 	var currentUserId = this.userId;
        // If current User is Admin, return all list to client
        if (currentUserId == Admin){
         return List.find();
      }
       else{
        // if current User is not Admin, then only return project which current user created 
 	return List.find({ createdBy: currentUserId});
      }
 })

But as you can see, if Admin want some specific user see a project, it's hard to useing this method.

I think we need to create a new box says "This project is for [my self] or 口", and drop all user name.
when server create a new project, it will add a new element like "toUser: Ming"
then the above function will change to something like:
var currentUserId = this.userId;
var currentUser = findUserName(currentUserId);
return List.find({ toUser: currentUser})

**Comment by cmojarro** [comment]
there is an atmosphere package that is available

https://atmospherejs.com/?q=roles

alanning:roles

**Comment by tvledesign** [comment]
I'm taking a look right now.

**Comment by tvledesign** [comment]
Okay so we can discuss more about this Wednesday if we don't have contact by then but so far it looks like you guys got the basic account system nailed down?

What I've noticed:

1. I can create an account
2. I can create a task
3. I can mark the task as complete

If I log out 
1. I won't see the task no more
2. I can't create tasks

So I assume this is working. The only thing we need left really is to assign roles correct? @cmojarro @bill668 

How are mock ups coming along @Claytonhaddock ?

**Comment by tvledesign** [comment]
Yeah I'm looking at the code and everything looks like how I would go about it. Nice job guys!

**Comment by bill668** [comment]
Finish some Update:
1. Now there's a build-in admin account : username: admin@admin.com  password: admin
Admin can view all element from List.

2. Fix some bugs

3. I create a new element 'toUser' inside task, and I changed the return List method from server to client which is replace [List.find({ created: currentUserId});]  by [List.find({ toUser: currentUserId});]
So in the future we can ask client: you create this project for who?
